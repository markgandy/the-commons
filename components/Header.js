import React, { PropTypes } from 'react'
import Link from 'next/link'
import classNames from 'classnames'

const links = [
  { href: '/', text: 'Home' },
  { href: '/about', text: 'About' },
  { href: '/bees', text: 'Bees' },
  { href: '/awards', text: 'Awards' },
  { href: '/photos', text: 'Photos' }
]

const Header = ({ currentUrl }) => (
  <section className='hero is-primary is-bold'>
    <div className='hero-body'>
      <div className='container is-fluid'>
        <h1 className='title'>
          The Commons
        </h1>
        <h2 className='subtitle'>
        Sustainable city living
        </h2>
        <nav className='nav has-shadow'>
          <div className='nav-left'>
            {links.map(link => (
              <Link href={link.href} key={link.href}>
                <a className={classNames('nav-item', 'is-tab', { 'is-active': currentUrl === link.href })}>
                  {link.text}
                </a>
              </Link>
            ))}
          </div>
        </nav>
      </div>
    </div>
  </section>
)

Header.propTypes = {
  currentUrl: PropTypes.string.isRequired
}

export default Header
