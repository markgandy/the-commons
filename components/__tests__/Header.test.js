import React from 'react'
import Header from '../Header'
import renderer from 'react-test-renderer'

test('Home page is active', () => {
  const component = renderer.create(
    <Header currentUrl='/' />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('About page is active', () => {
  const component = renderer.create(
    <Header currentUrl='/about' />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Awards page is active', () => {
  const component = renderer.create(
    <Header currentUrl='/awards' />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Bees page is active', () => {
  const component = renderer.create(
    <Header currentUrl='/bees' />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Photos page is active', () => {
  const component = renderer.create(
    <Header currentUrl='/photos' />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
