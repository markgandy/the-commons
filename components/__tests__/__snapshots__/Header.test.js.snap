exports[`test About page is active 1`] = `
<section
  className="hero is-primary is-bold">
  <div
    className="hero-body">
    <div
      className="container is-fluid">
      <h1
        className="title">
        The Commons
      </h1>
      <h2
        className="subtitle">
        Sustainable city living
      </h2>
      <nav
        className="nav has-shadow">
        <div
          className="nav-left">
          <a
            className="nav-item is-tab"
            href="/"
            onClick={[Function]}>
            Home
          </a>
          <a
            className="nav-item is-tab is-active"
            href="/about"
            onClick={[Function]}>
            About
          </a>
          <a
            className="nav-item is-tab"
            href="/bees"
            onClick={[Function]}>
            Bees
          </a>
          <a
            className="nav-item is-tab"
            href="/awards"
            onClick={[Function]}>
            Awards
          </a>
          <a
            className="nav-item is-tab"
            href="/photos"
            onClick={[Function]}>
            Photos
          </a>
        </div>
      </nav>
    </div>
  </div>
</section>
`;

exports[`test Awards page is active 1`] = `
<section
  className="hero is-primary is-bold">
  <div
    className="hero-body">
    <div
      className="container is-fluid">
      <h1
        className="title">
        The Commons
      </h1>
      <h2
        className="subtitle">
        Sustainable city living
      </h2>
      <nav
        className="nav has-shadow">
        <div
          className="nav-left">
          <a
            className="nav-item is-tab"
            href="/"
            onClick={[Function]}>
            Home
          </a>
          <a
            className="nav-item is-tab"
            href="/about"
            onClick={[Function]}>
            About
          </a>
          <a
            className="nav-item is-tab"
            href="/bees"
            onClick={[Function]}>
            Bees
          </a>
          <a
            className="nav-item is-tab is-active"
            href="/awards"
            onClick={[Function]}>
            Awards
          </a>
          <a
            className="nav-item is-tab"
            href="/photos"
            onClick={[Function]}>
            Photos
          </a>
        </div>
      </nav>
    </div>
  </div>
</section>
`;

exports[`test Bees page is active 1`] = `
<section
  className="hero is-primary is-bold">
  <div
    className="hero-body">
    <div
      className="container is-fluid">
      <h1
        className="title">
        The Commons
      </h1>
      <h2
        className="subtitle">
        Sustainable city living
      </h2>
      <nav
        className="nav has-shadow">
        <div
          className="nav-left">
          <a
            className="nav-item is-tab"
            href="/"
            onClick={[Function]}>
            Home
          </a>
          <a
            className="nav-item is-tab"
            href="/about"
            onClick={[Function]}>
            About
          </a>
          <a
            className="nav-item is-tab is-active"
            href="/bees"
            onClick={[Function]}>
            Bees
          </a>
          <a
            className="nav-item is-tab"
            href="/awards"
            onClick={[Function]}>
            Awards
          </a>
          <a
            className="nav-item is-tab"
            href="/photos"
            onClick={[Function]}>
            Photos
          </a>
        </div>
      </nav>
    </div>
  </div>
</section>
`;

exports[`test Home page is active 1`] = `
<section
  className="hero is-primary is-bold">
  <div
    className="hero-body">
    <div
      className="container is-fluid">
      <h1
        className="title">
        The Commons
      </h1>
      <h2
        className="subtitle">
        Sustainable city living
      </h2>
      <nav
        className="nav has-shadow">
        <div
          className="nav-left">
          <a
            className="nav-item is-tab is-active"
            href="/"
            onClick={[Function]}>
            Home
          </a>
          <a
            className="nav-item is-tab"
            href="/about"
            onClick={[Function]}>
            About
          </a>
          <a
            className="nav-item is-tab"
            href="/bees"
            onClick={[Function]}>
            Bees
          </a>
          <a
            className="nav-item is-tab"
            href="/awards"
            onClick={[Function]}>
            Awards
          </a>
          <a
            className="nav-item is-tab"
            href="/photos"
            onClick={[Function]}>
            Photos
          </a>
        </div>
      </nav>
    </div>
  </div>
</section>
`;

exports[`test Photos page is active 1`] = `
<section
  className="hero is-primary is-bold">
  <div
    className="hero-body">
    <div
      className="container is-fluid">
      <h1
        className="title">
        The Commons
      </h1>
      <h2
        className="subtitle">
        Sustainable city living
      </h2>
      <nav
        className="nav has-shadow">
        <div
          className="nav-left">
          <a
            className="nav-item is-tab"
            href="/"
            onClick={[Function]}>
            Home
          </a>
          <a
            className="nav-item is-tab"
            href="/about"
            onClick={[Function]}>
            About
          </a>
          <a
            className="nav-item is-tab"
            href="/bees"
            onClick={[Function]}>
            Bees
          </a>
          <a
            className="nav-item is-tab"
            href="/awards"
            onClick={[Function]}>
            Awards
          </a>
          <a
            className="nav-item is-tab is-active"
            href="/photos"
            onClick={[Function]}>
            Photos
          </a>
        </div>
      </nav>
    </div>
  </div>
</section>
`;
